app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaPresenter.java

```diff
@@ -178,8 +178,14 @@ private void checkNearbyPlaces(final UploadItem uploadItem) {
      * @param uploadItemIndex
      */
     @Override
-    public void verifyImageQuality(int uploadItemIndex, LatLng inAppPictureLocation) {
-      final UploadItem uploadItem = repository.getUploads().get(uploadItemIndex);
+    public boolean verifyImageQuality(int uploadItemIndex, LatLng inAppPictureLocation) {
+      final List<UploadItem> uploadItems = repository.getUploads();
+      if (uploadItems.size()==0) {
+          view.showProgress(false);
+          view.showMessage("Upload Failed",R.color.color_error);
```

Comment:
Please show a more technically detailed error message, explaining what is the actual issue. It is OK if users do not understand it fully.

Reply:
Thanks for your review! I changed it to ``view.showMessage("An out-of-sync of the number of Upload Media Detail Fragment and upload items happens.",R.color.color_error);``.

Reply:
How about this?

"Internal error: Zero upload items received by the Upload Media Detail Fragment. Sorry, please upload again."

Please also add a comment saying that internationalisation is not needed because this is an internal error message.

Thanks!

Reply:
Great idea! I've completed it